### ISSUE TITLE
Style not applied for my Custom button

### ISSUE THREAD

**Issue by samyraj42** [AvaloniaUI/Avalonia#17545]
### Describe the bug

I have written code for my custom button. I have created a style for that.
But that style was not applied to my custom button

### To Reproduce

Step 1:

internal class CustomButton : Button
{
    public CustomButton()
    {
        //Background = Brushes.LightBlue;
        //this.Classes.Add("red");
        // Set default styles or properties
        HorizontalAlignment = HorizontalAlignment.Stretch;
        VerticalAlignment = VerticalAlignment.Stretch;
    }

    // ImageSource Dependency Property
    public static readonly StyledProperty<Bitmap?> ImageSourceProperty =
        AvaloniaProperty.Register<CustomButton, Bitmap?>(nameof(ImageSource));

    public Bitmap? ImageSource
    {
        get => GetValue(ImageSourceProperty);
        set => SetValue(ImageSourceProperty, value);
    }

    // ButtonText Dependency Property
    public static readonly StyledProperty<string> ButtonTextProperty =
        AvaloniaProperty.Register<CustomButton, string>(nameof(ButtonText));

    public string ButtonText
    {
        get => GetValue(ButtonTextProperty);
        set => SetValue(ButtonTextProperty, value);
    }
    protected override Type StyleKeyOverride => typeof(Button);
    // Override Control Template to include image and text
    protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
    {
         base.OnApplyTemplate(e);
        // Optionally, handle template-specific logic here
    }

}

Step2:
<Style Selector="local|CustomButton">
	<Setter Property="Background" Value="Red"/>
</Style>

Setp3:
<local:CustomButton 
					Width="150" Height="75" />

### Avalonia version

11.1.0

### OS

Windows

**Comment by timunie** [comment]
Why do you do a stylekeyoverride if you want to target your custom button? That's wrong usage here. 

In case you wanted to base on button style, you need to use ControlTheme with BasedOn set accordingly.

**Comment by maxkatz6** [comment]
Yes, by using StyleKeyOverride you told styling subsystem, that it should use "Button" as a style key.
Which is why "local|CustomButton" is not matching anymore.
